### ISSUE TITLE
Appimage from the terminal, and relative paths

### ISSUE THREAD

**Issue by Alatius** [fontforge/fontforge#5067]
Now that an Appimage is the preferred way to distribute FontForge for Linux, how do you use it as a command line tool (specifically to run scripts)? I figured that I could simply link to the latest appimage:
`sudo ln -s ~/Appimages/FontForge-2022-03-08-582bd41-x86_64.AppImage /usr/local/bin/fontforge`
However, when I try to execute a script in the current directory, I get file not found:
```
$ fontforge -script script.pe 
Copyright (c) 2000-2022. See AUTHORS for Contributors.
 License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 with many parts BSD <http://fontforge.org/license.html>. Please read LICENSE.
 Version: 20220308
 Based on sources from 2022-03-08 10:15 UTC-ML-TtfDb-D-GDK3.
 Based on source from git with hash: 582bd41a9bf04326300fc02a677fe3610d6d3ccd
script.pe: No such file
```
I guess that the executable is run within the temporary mounted directory, and that the relative path is interpreted relative to that dir? How should this be handled?
(I should mention that `fontforge -script \`pwd\`/script.pe` works, insofar that the script is executed. But in my use case, the script itself includes relative paths, and breaks for that reason.)

**Comment by ctrlcctrlv** [comment]
Hmm. Interesting problem. @probonopd, can I ask you for a suggestion maybe?

**Comment by probonopd** [comment]
You need to provide absolute paths to the FontForge AppImage (since the `AppRun` inside it does a `chdir()`/`cd` to `<AppDir>/usr/`, iirc.

Luckily this is easy: instead of running `fontforge -script script.pe` you need to run `fontforge -script $(readlink -f script.pe)`.

In case this is not desirable, one would need to use a different AppRun` which does not do a `chdir()`/`cd` to `<AppDir>/usr/`.

**Comment by ge9** [comment]
I want to make my font build script work in both Linux and Windows, but this problem breaks the compatibility of the script.
The AppImage should see the current directory.

**Comment by probonopd** [comment]
Can you pass absolute rather than relative paths to FontForge using your build script? That'd solve it without having to change the FontForge AppImage.

**Comment by ge9** [comment]
Yes, `readlink -f` works for Linux version (thank you for the information). Windows version works with relative paths. But Cygwin's `readlink` returns Cygwin path, which doesn't suite Windows fontforge.
Now I decided to use a variable `GET_SCRIPT_PATH_FOR_FONTFORGE` in my Makefile and set it `echo` in Windows and `readlink -f` in Linux (detecting OS by environment variables). Then I can run fontforge like `fontforge -script $($GET_SCRIPT_PATH_FOR_FONTFORGE script.pe)` in both OSs.
It does work. FontForge AppImage is a cool solution for building fonts (especially because it works without root privilege). But I hope FontForge AppImage will be aware of the original current directory, in future.
